### ISSUE TITLE
DisplayMagician throws Exception and won't start

### ISSUE THREAD

**Issue by m0ellemeister** [terrymacdonald/DisplayMagician#7]
**Describe the bug**
When I try to start DisplayMagician, a Fatal Error pops up: "Type initialization Exception caused by: DisplayMagicianShared.ProfileRepository" This happens regardless if NVIDIA Surround is enabled or not.

Maybe this is caused somehow by new NVIDIA Drivers? Just a wild guess. I updated my drivers to Version 466.11

**To Reproduce**
Steps to reproduce the behaviour:
1. Install DisplayMagician 1.0.4
3. App throws Exception at Start

**Expected behaviour**
DisplayMagician should start without errors.

**Screenshots**
![image](https://user-images.githubusercontent.com/3107650/115111698-cfc00e80-9f81-11eb-9298-d5a11e22c2c7.png)

**Reporting Logs**
If this Logs aren't enough, I will collect the logs later. Here's what I found in DisplayMagician.log

```
2021-04-17 13:28:52.3568|INFO|DisplayMagician.Program|Starting DisplayMagician v1.0.4.0
2021-04-17 13:28:52.4098|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Uplay not installed.
2021-04-17 13:28:52.4098|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/SteamLibrary: Steam library is installed in C:\Program Files (x86)\Steam. Found C:\Program Files (x86)\Steam\steam.exe
2021-04-17 13:28:52.4098|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Loading Installed Steam Games
2021-04-17 13:28:52.6918|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found original steam library C:\Program Files (x86)\Steam
2021-04-17 13:28:52.6918|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found additional steam library E:\SteamLibrary
2021-04-17 13:28:52.7098|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found 17 installed Steam games
2021-04-17 13:28:52.7098|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Loaded all Installed Steam Games (found 17)
2021-04-17 13:28:52.7779|ERROR|DisplayMagician.Program|Program/StartUpNormally top level exception: Der Typeninitialisierer für "DisplayMagicianShared.ProfileRepository" hat eine Ausnahme verursacht.:    bei DisplayMagicianShared.ProfileRepository.get_AllProfiles()
   bei DisplayMagician.UIForms.MainForm.RefreshNotifyIconMenus() in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:Zeile 214.
   bei DisplayMagician.UIForms.MainForm..ctor(Form formToOpen) in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:Zeile 34.
   bei DisplayMagician.Program.StartUpApplication() in H:\vscode-projects\DisplayMagician\DisplayMagician\Program.cs:Zeile 475. - NvAPIWrapper.Native.Exceptions.NVIDIAApiException: NVAPI_ERROR
   bei NvAPIWrapper.Native.GPUApi.GetBoardInfo(PhysicalGPUHandle gpuHandle)
   bei NvAPIWrapper.GPU.PhysicalGPU.get_Board()
   bei DisplayMagicianShared.ProfileRepository.GenerateProfileDisplayIdentifiers() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 810.
   bei DisplayMagicianShared.ProfileItem.Equals(ProfileItem other) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileItem.cs:Zeile 338.
   bei DisplayMagicianShared.ProfileRepository.IsActiveProfile(ProfileItem profile) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 589.
   bei DisplayMagicianShared.ProfileRepository.LoadProfiles() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 648.
   bei DisplayMagicianShared.ProfileRepository..cctor() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 80.
```

**Enviroment (please complete the following information):**
 - Windows Version: Windows 10 Home 20H2 Build 19042.928
 - NVIDIA Driver 466.11 (RTX 2080ti)
 - DisplayMagician Version 1.0.4

**Comment by terrymacdonald** [comment]
Hmmmm. This appears to be something in the profilerepository part of the codebase. I'll take a look at the line numbers to see if there is something I can see happening there. Are you able to please go to your AppData\DisplayMagician|Profiles folder and grab the DisplayProfiles_1.0.json within the Profile, and add them to this issue please? I'd like to see what your Display Profiles are so I can tell whether there is an issue with the file itself.

Could you also please open DisplayMagician, click on the settings button, and set the Log Level to 'TRACE'. Then restart DisplayMagician and then attach the full DisplayMagician.log to this issue too. Both of those things will hopefully give me the visibility that I need to troubleshoot this.

**Comment by m0ellemeister** [comment]
I'm not able to start DisplayMagician at because of the Exception shown on the first Screenshot of my initial post. Nut I set the LogLevel in C:\User\MyUser\AppData\Local\DisplayMagician\Settings_1.0.json to "Trace" like this:

```json
{
  "StartOnBootUp": true,
  "MinimiseOnStart": true,
  "LogLevel": "Trace"
}
```

The DisplayMagician.log file likes this after setting LogLevel to "Trace":

```
2021-04-19 15:50:10.9099|INFO|DisplayMagician.Program|Starting DisplayMagician v1.0.4.0
2021-04-19 15:50:10.9619|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/SteamLibrary: Steam library is installed in C:\Program Files (x86)\Steam. Found C:\Program Files (x86)\Steam\steam.exe
2021-04-19 15:50:10.9619|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Loading Installed Steam Games
2021-04-19 15:50:10.9619|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Uplay not installed.
2021-04-19 15:50:11.3320|ERROR|DisplayMagician.Program|Program/StartUpNormally top level exception: Der Typeninitialisierer für "DisplayMagicianShared.ProfileRepository" hat eine Ausnahme verursacht.:    bei DisplayMagicianShared.ProfileRepository.get_AllProfiles()
   bei DisplayMagician.UIForms.MainForm.RefreshNotifyIconMenus() in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:Zeile 214.
   bei DisplayMagician.UIForms.MainForm..ctor(Form formToOpen) in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:Zeile 34.
   bei DisplayMagician.Program.StartUpApplication() in H:\vscode-projects\DisplayMagician\DisplayMagician\Program.cs:Zeile 475. - NvAPIWrapper.Native.Exceptions.NVIDIAApiException: NVAPI_ERROR
   bei NvAPIWrapper.Native.GPUApi.GetBoardInfo(PhysicalGPUHandle gpuHandle)
   bei NvAPIWrapper.GPU.PhysicalGPU.get_Board()
   bei DisplayMagicianShared.ProfileRepository.GenerateProfileDisplayIdentifiers() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 810.
   bei DisplayMagicianShared.ProfileItem.Equals(ProfileItem other) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileItem.cs:Zeile 338.
   bei DisplayMagicianShared.ProfileRepository.IsActiveProfile(ProfileItem profile) in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 589.
   bei DisplayMagicianShared.ProfileRepository.LoadProfiles() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 648.
   bei DisplayMagicianShared.ProfileRepository..cctor() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:Zeile 80.
2021-04-19 15:50:11.3820|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found original steam library C:\Program Files (x86)\Steam
2021-04-19 15:50:11.3820|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found additional steam library E:\SteamLibrary
2021-04-19 15:50:11.3870|INFO|DisplayMagician.GameLibraries.SteamLibrary|SteamLibrary/LoadInstalledGames: Found 19 installed Steam games
2021-04-19 15:50:11.3870|INFO|DisplayMagician.Program|Program/LoadGamesInBackground: Loaded all Installed Steam Games (found 19)
```
Heres' my DisplayProfiles_1.0.json:

```json
[
  {
    "UUID": "d9bbfd0c-ef21-4dfc-8829-fa7b5dec43a0",
    "Name": "Normaler Desktop",
    "Paths": [
      {
        "PixelFormat": "PixelFormat32Bpp",
        "Position": "1920, -1",
        "Resolution": "1920, 1080",
        "SourceId": 0,
        "TargetDisplays": [
          {
            "DevicePath": "\\\\?\\DISPLAY#GSM76F5#5&1d410d75&0&UID37121",
            "DisplayName": "32inch LG FHD",
            "FrequencyInMillihertz": 60000,
            "Rotation": "Identity",
            "Scaling": "Identity",
            "ScanLineOrdering": "Progressive",
            "SurroundTopology": null
          }
        ]
      },
      {
        "PixelFormat": "PixelFormat32Bpp",
        "Position": "-1920, 0",
        "Resolution": "1920, 1080",
        "SourceId": 1,
        "TargetDisplays": [
          {
            "DevicePath": "\\\\?\\DISPLAY#GSM76F5#5&1d410d75&0&UID37120",
            "DisplayName": "32inch LG FHD",
            "FrequencyInMillihertz": 60000,
            "Rotation": "Identity",
            "Scaling": "Identity",
            "ScanLineOrdering": "Progressive",
            "SurroundTopology": null
          }
        ]
      },
      {
        "PixelFormat": "PixelFormat32Bpp",
        "Position": "0, 0",
        "Resolution": "1920, 1080",
        "SourceId": 2,
        "TargetDisplays": [
          {
            "DevicePath": "\\\\?\\DISPLAY#GSM76F5#5&1d410d75&0&UID37124",
            "DisplayName": "32inch LG FHD",
            "FrequencyInMillihertz": 60000,
            "Rotation": "Identity",
            "Scaling": "Identity",
            "ScanLineOrdering": "Progressive",
            "SurroundTopology": null
          }
        ]
      }
    ],
    "SavedProfileIconCacheFilename": "C:\\Users\\smoel\\AppData\\Local\\DisplayMagician\\Profiles\\profile-d9bbfd0c-ef21-4dfc-8829-fa7b5dec43a0.ico",
    "ProfileDisplayIdentifiers": [
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|DisplayPort|2147881090",
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|DisplayPort|2147881095",
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|HDMI|2147881091"
    ],
    "ProfileBitmap": "iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAYAAABccqhmAAAABGdBTUEAALGPC/xhBQAAAAlwSFlzAAAOwgAADsIBFShKgAAABX9JREFUeF7t2l9ollUAx/GBFIEUXgQSQRFEQdA2HXQ1UNCB5Eb6Dsa6WIbkTaAXVuoY/r1oFF2oSBKOMGNliIl3shtBEMolI6dSEN5ETEGFSUzYYG/PwXdI+Cywc+LQ+3w+8MWbOdjxPb/32VwLAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAANm0t7d39fT07O/t7f1CUnO1cePGT9va2ja0trYubVz5h2q12qGRkZErFy9evDY5OfmzpOZqfHx88uTJk79s2rRp9G8jEN75w+Wfn5//QVJzNzY2Ntnd3f1h4/q3tKxfv37w/Pnz18s+WFJzFZ4GwrcEjetvAKQqZQCkCmcApApnAKQKZwD0v+7GjRsTWrypqanLZee20GMPwK1bt37at2/fncHBwWkt3rlz534tO7+Fhr88+ftbew5Oa/E+OjRys+zsFipc3blz5/29e/fOqLyhoaF7/zQCjz0AYVXCJz527Fhd5Q0PD9dPnTp1o+z8Fnrvs2O3l3zzR73l+2ktUhiBsrNb6MKFC9cPHDhQ+m+gB4W7Gu5s2fmFDMB/kAFIkwGIzwBkyACkyQDEZwAyZADSZADiMwAZMgBpMgDxGYAMGYA0GYD4DECGDECaDEB8BiBDBiBNBiA+A5AhA5AmAxCfAciQAUiTAYjPAGTIAKTJAMRnADJkANJkAOIzABkyAGkyAPEZgAwZgDQZgPgMQIYMQJoMQHwGIEMGIE0GID4DkCEDkCYDEJ8ByJABSJMBiM8AZMgApMkAxGcAMmQA0mQA4jMAGTIAaTIA8RmADBmANBmA+AxAhgxAmgxAfAYgQwYgTQYgPgOQIQOQJgMQnwHIkAFIkwGIzwBkyACkyQDEZwAyZADSZADiMwAZMgBpMgDxGYAMGYA0GYD4DECGDECaDEB8BiBDBiBNBiA+A5AhA5AmAxCfAciQAUiTAYjPAGTIAKTJAMRnADJkANJkAOJLPgBTU1OXh4aG7oVPrPIGBwfvnz179rey81vogyNf3ewY/nam45NTWqR3Pz5yp+zsFipcLc56Wou3e/fuu+HOlp1f6LEHIBQ+YVgVLd7MzMylsrNbaHp6erzs7+lhd+/eHS87O6XrXw2ApObIAEgVzgBIFc4ASBXOAEgV7pEBaG9vf/PEiRPXyj5YUnMV3uzDm37j+re0dHR0PNHX1zcSRmBiYuJKWAhJzdfp06evDwwMnGlra1vWuP4PhBEITwK1Wu1QeDyQ1Hx1dna+3draurRx7QEAAMigXq8/NT8//3RF8j0oBMXFf7K4EK+OjY2tP3r06JYqNDo62l98za+HMWgcA1RPcfmXFJfgtW3btn3X19c3t2HDhkrU29s7t3nz5onbt2+/EZ58GscB1VK8+JcdPnz4/VqtNrdy5crZFStWVKbu7u65Xbt2HSwG8PnGcUC1FC/+5eESrFu3bq7skjRzq1evnt26deuZ4gxebBwHVEvx4n9ux44dn3d1dZVekmbOAFB5BsAAUGEGwABQYfV6/dn9+/fv6enpqdzPANasWTO7ffv2r4szeKFxHFAt4b/ALl26tHZgYODm2rVrZzs7OyvRqlWrZvv7+/8Mvw8QRrBxHFA9xSPw8jAC4d1wy5YtP1ah8HsPx48ff6f42l9uHANUV3ERngmPwsWfr1Skl7zzAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAo1pa/gK1drpJg7SAOQAAAABJRU5ErkJggg==",
    "ProfileTightestBitmap": "iVBORw0KGgoAAAANSUhEUgAAAQAAAAAzCAYAAAB469bRAAAABGdBTUEAALGPC/xhBQAAAAlwSFlzAAAOwgAADsIBFShKgAAAArxJREFUeF7t3cGKUmEYh3H30bILqAsQQVq3cFXuPHYpIqJiugtazC4GoU3ugtxEFyAITVJWNiFBLtzoSnISB2dA8wPbfX8hGHV8v+eF3+bIgTmv53sWbiZ2ExOPx+8kk8mnAPYjkUjc2xy/w08URaeNRuNXs9n8CWD3stnsu83xO/y4P2Y8Hn9aLpcfAOxeJpPpbI7f4YcAAPtFAICAEQAgYAQACNhRBWA4HH4eDAbdEI1Go61hdJ/77guBey98O/kn5N04i8XizLcX52gC0O/3vxSLxVmlUpmHaP3sFyoC7vrj6suL5PM38xA9qtZn7v3w7WY6nXZKpdJv305DkM/nL9vt9rlvN87RBKDb7X6rVqtX9Xp9FSL3Zbqa+3bjrruDEHs7XYXo/ov3V+798O1mMpl0yuXyH99OQ1Cr1VatVuuHbzcOATgSBEAjABoBMIIAaARAIwBGEACNAGgEwAgCoBEAjQAYQQA0AqARACMIgEYANAJgBAHQCIBGAIwgABoB0AiAEQRAIwAaATCCAGgEQCMARhAAjQBoBMAIAqARAI0AGEEANAKgEQAjCIBGADQCYAQB0AiARgCMIAAaAdAIgBEEQCMAGgEwggBoBEAjAEYQAI0AaATACAKgEQCNABhBADQCoBEAIwiARgA0AmAEAdAIgEYAjCAAGgHQCIARBEAjABoBMIIAaARAIwBGEACNAGgEwAgCoBEAjQAYQQA0AqARACMIgEYANFMByOVy1+6BQuT+z/vWANReXd49aa1C9ODZ6+ttASgUCjPfTkOwPjM2AjCfzz+6BwlVu90+XywWZ77duOvuc999oXDvh283znq+++4JhYugby/O0QQAwM0jAEDACAAQMAIABIwAAAG7dQFwv2b3er2vAHbvVgUglUo9jKLoFMB+pNPpJ5vjxzAMwzAM818Ti/0FAs6nKUkJccAAAAAASUVORK5CYII="
  },
  {
    "UUID": "c8f84c61-1ad3-465d-90e6-294b4b0cf7f2",
    "Name": "NVIDIA Surround",
    "Paths": [
      {
        "PixelFormat": "PixelFormat32Bpp",
        "Position": "0, 0",
        "Resolution": "5760, 1080",
        "SourceId": 0,
        "TargetDisplays": [
          {
            "DevicePath": "\\\\?\\DISPLAY#GSM76F5#5&1d410d75&0&UID37888",
            "DisplayName": "NV Surround",
            "FrequencyInMillihertz": 59940,
            "Rotation": "Identity",
            "Scaling": "Identity",
            "ScanLineOrdering": "Progressive",
            "SurroundTopology": {
              "AcceleratePrimaryDisplay": true,
              "ApplyWithBezelCorrectedResolution": false,
              "BaseMosaicPanoramic": false,
              "ColorDepth": 32,
              "Columns": 3,
              "Displays": [
                {
                  "DisplayId": 2147881091,
                  "DisplayName": "32inch LG FHD",
                  "Overlap": "0, 0",
                  "PixelShift": "NoPixelShift",
                  "Rotation": "Identity"
                },
                {
                  "DisplayId": 2147881095,
                  "DisplayName": "32inch LG FHD",
                  "Overlap": "0, 0",
                  "PixelShift": "NoPixelShift",
                  "Rotation": "Identity"
                },
                {
                  "DisplayId": 2147881090,
                  "DisplayName": "32inch LG FHD",
                  "Overlap": "0, 0",
                  "PixelShift": "NoPixelShift",
                  "Rotation": "Identity"
                }
              ],
              "DriverReloadAllowed": false,
              "Frequency": 60,
              "ImmersiveGaming": true,
              "Resolution": "1920, 1080",
              "Rows": 1
            }
          }
        ]
      }
    ],
    "SavedProfileIconCacheFilename": "C:\\Users\\smoel\\AppData\\Local\\DisplayMagician\\Profiles\\profile-c8f84c61-1ad3-465d-90e6-294b4b0cf7f2.ico",
    "ProfileDisplayIdentifiers": [
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|DisplayPort|2147881090",
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|DisplayPort|2147881095",
      "NVIDIA|TU102-A|161|Serial 00-00-00-00-00-00-00-00-00-00-00-00-00-00-00-00|TSMC|2304|Discrete|HDMI|2147881091"
    ],
    "ProfileBitmap": "iVBORw0KGgoAAAANSUhEUgAAAQAAAAEACAYAAABccqhmAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAP+SURBVHhe7dpPSKR1HMfxoQgKCYq6LAXRNWJm3DkHHvzDqIjOsCKCCUIkoVASZve9mIfqogQexCi8jRcPaeA1hG0z+8cePHix8xIdHHB6fuwTCzUsuEQrz/f1gjd7GYR58PsZ190SAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAE9MpVLpq9VqtyQVs2q1Opz9+Ux+8g9NTEx8sr6+/kOr1bonqZhtb2//PD09/XW5XO7JT//BJ386/svLy+8kFbu9vb2f6vX6Qn7+pdLg4OD7BwcHv3Z7saRidXJyctxsNr/Iz79UGhkZ+fjw8PCXbi+WVKwMgBQ4AyAFzgBIgXusATg/P79zenp6V9L1Lt1qtxv+uysPQPqCC7fr99/9tPanpOtdutVHjcCVByCtSvrC731T6ki63qVbTTfb7ZZTBkAqcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAJnAKTAGQApcAZACpwBkAL3nw/A+fn5nYXb9fvpC0u63qVbPTs7+77bLaeuPACpNAJpVSRd7x51/KnHGgBJxcgASIEzAFLgDIAUOAMgBe5fA1CtVgc2Nzd/7PZiScVqf3//ZHR09MP8/B9oNBqfpxFotVr3JBWznZ2d32ZnZ78ql8s9+ek/lH4SqNVqtyQVs0qlMt71+AEAAPjfXF5e9mQ9H6VOp/NU/tYhruwYnst6Y2tr6+2NjY13opS950o2Ai/mjwHiyQ7g6ewQ3pyfn/92cnKyPT4+HqJms9mem5u7e3x8/Fb2/v1GmpiyAXhhbW3tg0aj0b558+ZFb29vmIaHh9vLy8vr2QC8kj8OiCX75r+xsrLy2dDQULvbkRS5vr6+i8XFxVb2DF7LHwfEkgYgfQoODAx0PZIiZwAIzwAYAAIzAAaAwLJv/pdWV1c/GhsbC/c7gP7+/oulpaUvs2fwav44IJZOp/Ns+qewmZmZ3+v1ejsdRfppoOil9zk1NfXH7u7uRBrB/HFAPNkIvHx0dNSf/ioQqfz4X88fA8SVHUL677E3guWTHwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA/qFU+gsJ2JLvac0WmAAAAABJRU5ErkJggg==",
    "ProfileTightestBitmap": "iVBORw0KGgoAAAANSUhEUgAAAQAAAAAzCAYAAAB469bRAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGVSURBVHhe7dSxqpIBAIZh14aGiPbuQOGnK3DL7ag1RUNB0IXEGYLWGoIGwS0bImgLQagOEWE1FDk7OXpw0c4Pjg4KRz36PR88l/C9lctYrVa7VRRFG9iNarV6e3m//a/dbnc7nc6/Xq/3B9i+i899Wt5v/2u1Wu/H4/G3+Xz+Gdi+k5OTs+X99j8BgN0SAAgmABBMACDYQQVgNBp9BzYzm82+rPpT6WACcLGfj54V509eFFNgPY9Pi/N3H7p/V32qdDAB6Pf7vx+8vL54+rECrOnhm2uL7tvXo1WfKgkAHDEBgGACAMEEAIIJAAQTAAgmABBMACCYAEAwAYBgAgDBBACCCQAEEwAIJgAQTAAgmABAMAGAYAIAwQQAggkABBMACCYAEEwAIJgAQDABgGACAMEEAIIJAAQTAAh2VAG4d3pzUUYAWM/95zeOIwDT6fRrGQFgM5PJ5GzVp0oHEwDg8gkABBMACCYAEEwAINiVC8BgMPg1HA5/ANt3pQJQr9fvNJvNV8BuNBqNu8v7mZmZmW20SuU/QX4d5nL8r9EAAAAASUVORK5CYII="
  }
]
```

Within my Profiles Folder I have 3 *.ico Files:
![image](https://user-images.githubusercontent.com/3107650/115247163-d76ce800-a126-11eb-986c-ce9c1a49c3e1.png)


I renamed C:\User\MyUser\AppData\Local\DisplayMagician\ to C:\User\MyUser\AppData\Local\DisplayMagician_broken\ and tried again to start DisplayMagician. Same error. Might there something in the Windows Registry I should delete? Or some where else?

**Comment by m0ellemeister** [comment]
Another Update. I think I found the reason for the Exception at startup, but I I'm not sure what's wrong.

I installed VisualStudio CE, cloned DisplayMagician Repo and tried to dive into the code.

 In ProfileRepository.cs there's the fuction call `myPhysicalGPU.Board.ToString()` around line 961. If `myPhysicalGPU.Board.ToString()` is commented out, DisplayMagician will start. 
Every call on `myPhysicalGPU.Board` will fail. I don't know why.
Also I don't know why aDisplayDevice.Output.OutputId.ToString() was commented out. I commented it in again and DisplayMagician still starts.

I will do some testing with my Build. If it seems OK I can provide a diff.

**Comment by terrymacdonald** [comment]
That all points towards a problem with the NVAPIwrapper library.  I'll try updating my Nvidia driver and see if I start getting problems myself. Great work @m0ellemeister!

**Comment by terrymacdonald** [comment]
OK, so I was running NVIDIA 465.89 Display Driver here and it was working fine. I updated to 466.11, and even after a reboot DisplayMagician v1.0.4 would start correctly without any issues. 

Did you create the Display Profiles in DisplayMagician v1.0.4? Or in an earlier version of DisplayMagician? I'd like to know this to see if that may have caused any issues....

I'm a little at a loss as to why skipping that `myPhysicalGPU.Board.ToString()` works. The description you've given indicates that NvAPIWrapper.net library is causing an exception when the accessing that specific item on your computer. That same command appears to have worked for you previously, as it is used to generate the  "ProfileDisplayIdentifiers" shown in the profile JSON you posted.  The ProfileDisplayIdentifiers are used to detect whether a profile can be used or not at any particular time, so its critical that they match exactly, otherwise the Display Profiles won't be able to be used.

I will be able to avoid the NvAPIWrapper exception from stopping DisplayMagician from running, by replacing any failed NVIDIA lookups with some default entries, but that unfortunately won't work in every situation. If NvAPIWrapper fails to access any information consistently then this will work, but if the lookup intermittently fails then it will still cause issues matching the ProfileDisplayIdentifiers. I think that's the best I can do for now.

I'm quite close to completing a new release to add Origin game support, so I'll tag this as a fix for that version.

**Comment by m0ellemeister** [comment]
I created the Display Profiles in DisplayMagician 1.0.4.

I was trying to get to the Exception the NvAPIWrapper.Native.GPUApi.GetBoardInfo(PhysicalGPUHandle gpuHandle) call gets from the NVIDIA API. I wasn't successfull. I was trying to get Message and Status from https://github.com/falahati/NvAPIWrapper/blob/2a41eb4d6b1bceff5dbb33a3b5ea70bd582214a7/NvAPIWrapper/Native/Exceptions/NVIDIAApiException.cs. But I only got the calling function, I was not able to get the Response of the NVIDIA API through NvAPIWrapper.

Maybe something on my side is fishy. I installed Windows 10 eight days ago (after running Linux for over a decade). installed new NVIDIA Drivers as soon as NVIDIAs Desktop App notified my. I installed every Windows Update that was available including feature packs. Maybe the issue is related to my ROG Strix RTX 2080ti somehow. I can't remember exactly, but I think I installed ASUS ARMOURY CRATE and ASUS Aura Studio after DisplayMagician. Maybe ASUS changed something in the NVIDIA API in regards to their Aura RGB LED stuff ^^ just a guess.

I haven't found any issues related to my hardware in the Windows System Logs.

I will try to build an Exception Handling around these blocks of code:
```c#
                            // Create an array of all the important display info we need to record
                            string[] displayInfo = {
                                "NVIDIA",
                                myPhysicalGPU.ArchitectInformation.ShortName.ToString(),
                                myPhysicalGPU.ArchitectInformation.Revision.ToString(),
                                myPhysicalGPU.Board.ToString(),
                                myPhysicalGPU.Foundry.ToString(),
                                myPhysicalGPU.GPUId.ToString(),
                                myPhysicalGPU.GPUType.ToString(),
                                aDisplayDevice.Output.OutputId.ToString(),
                                aDisplayDevice.ConnectionType.ToString(),
                                aDisplayDevice.DisplayId.ToString(),
                            };
```
and write to the JSON Config file, how the Identifier is defined, what fields are given. At startup DisplayMagician could check the config for a definition of the identifier. That way DisplayMagician should have a well defined and stable Identifier. What do you think @terrymacdonald ?

**Comment by Horcon** [comment]
I also have this issue, and I also very recently installed Windows 10. I also have a 2080ti (Gigabyte) and use a 970 to use my second row of screens, 6 in total. The bottom 3 screens are what I'd use in surround, with the top being used for video, discord or guides. So it's not just you.

**Comment by terrymacdonald** [comment]
Thanks Horcon. I wonder if it's a problem with NVIDIA driver and 2080Ti
boards then? If it is, and the error happens reliably, then I think I can
work around it. I have some code in mind that will fill the missing
getboard lookup with a default value, which will still allow the
profileidentifier matching to work. You may need to recreate your display
profiles though... I'lll see what I can do.

Don't bother writing code to do a pull request m0ellemeister as I've almost
got a new build ready to go. I think I'll have it done and tested by the
weekend (depending on workload at $dayjob)

I think I can work around this issue. I'm very interested of anyone else
has a similar issue, and if they also have an NVIDIA RTX 2080Ti board.

Terry

**Comment by terrymacdonald** [comment]
OK! I've crafted a work around and it at least works for me. I don't have a RTX 2080Ti to test with :(. I only have an old RTX 1070, so I'll need some help from those with RTX 2080TI's to do some testing when I release DIsplayMagician v1.0.5 containing this fix.

I'm also trying to roll in another fix and a couple of new features, so you'll have to wait a few more days before the new release will appear.  I've included the commit message I posted for the fix to give you a background on what's changed:

Added exception protection around the video
card identifier errors to at least allow
DisplayMagician to kind of keep working even
if there are errors in the NVIDIA drivers. This fix
was brought about to get around problems with
NVIDIA RTX 2080Ti cards when using NVIDIA
driver 466.11. This fix should allow users with
cards that report data incorrectly to recreate new
DisplayProfiles, and then change all their shortcuts
to use those new Display Profiles. DisplayMagician
should then start working again. Fixes #7.

NOTE: This problem is a workaround for a broken
NVIDIA driver. Once NVIDIA find out and fix it, then
you will need to recreate your Display Profile again
and change all your shortcuts to the new Display
Profile in order for your shortcuts to work again.

**Comment by m0ellemeister** [comment]
I will test the new release!  It will be pleasure :-D

**Comment by joeymanson23** [comment]
I have the same issue when launching the application. But apparently a reboot solves the problem. But then it happens again.
I recon this started after NVIDIA drivers update.
![image](https://user-images.githubusercontent.com/81873707/115743495-a2140480-a389-11eb-8cbd-5d60f425b5b1.png)

Regards,
Joey

**Comment by terrymacdonald** [comment]
OK everyone. This *should* be fixed now within v1.0.5 which I've just released. I will close this issue for now, but I will reopen it if there are still problems matching this specific issue.

DisplayMagician should inform you of the new v1.0.5 when you next restart and open it.
